server/dex/dex.go

```diff
@@ -0,0 +1,254 @@
+
+		// Ensure the symbol is a recognized BIP44 symbol, and retrive its ID.
+		ID, found := dex.BipSymbolID(symbol)
+		if !found {
+			return nil, fmt.Errorf("asset symbol %q unrecognized", assetConf.Symbol)
+		}
+
+		// Double check the asset's network.
+		net, err := dex.NetFromString(assetConf.Network)
+		if err != nil {
+			return nil, fmt.Errorf("unrecognized network %s for asset %s",
+				assetConf.Network, symbol)
+		}
+		if cfg.Network != net {
+			return nil, fmt.Errorf("asset %q is configured for network %q, expected %q",
+				symbol, assetConf.Network, cfg.Network.String())
+		}
+
+		assetIDs[i] = ID
+	}
+
+	// Make an internal context and cancel function.
+	ctxDex, cancel := context.WithCancel(ctx)
+
+	// Create a MasterCoinLocker for each asset.
+	dexCoinLocker := coinlock.NewDEXCoinLocker(assetIDs)
+
+	// Start asset backends.
+	var dcrBackend *dcrasset.DCRBackend
+	lockableAssets := make(map[uint32]*swap.LockableAsset, len(cfg.Assets))
+	backedAssets := make(map[uint32]*asset.BackedAsset, len(cfg.Assets))
+	for i, assetConf := range cfg.Assets {
+		symbol := strings.ToLower(assetConf.Symbol)
+		ID := assetIDs[i]
+
+		// Create a new asset backend. An asset driver with a name matching the
+		// asset symbol must be available.
+		subsys := fmt.Sprintf("ASSET[%s]", symbol)
+		logger := slog.NewBackend(os.Stdout).Logger(subsys)
+		log.Infof("Starting asset backend %q...", symbol)
+		be, err := asset.Start(symbol, ctxDex, assetConf.ConfigPath, logger, cfg.Network)
+		if err != nil {
+			cancel()
+			return nil, fmt.Errorf("failed to start asset %q: %v", symbol, err)
+		}
+
+		if symbol == "dcr" {
+			var ok bool
+			dcrBackend, ok = be.(*dcrasset.DCRBackend)
+			if !ok {
+				cancel()
+				return nil, fmt.Errorf("dcr backend is invalid")
+			}
+		}
+
+		ba := &asset.BackedAsset{
+			Asset: dex.Asset{
+				ID:       ID,
+				Symbol:   symbol,
+				LotSize:  assetConf.LotSize,
+				RateStep: assetConf.RateStep,
+				FeeRate:  assetConf.FeeRate,
+				SwapConf: assetConf.SwapConf,
+				FundConf: assetConf.FundConf,
+			},
+			Backend: be,
+		}
+
+		backedAssets[ID] = ba
+		lockableAssets[ID] = &swap.LockableAsset{
+			BackedAsset: ba,
+			CoinLocker:  dexCoinLocker.AssetLocker(ID).Swap(),
+		}
+	}
+
+	// Create DEXArchivist with the pg DB driver.
+	pgCfg := &pg.Config{
+		Host:          cfg.DBConf.Host,
+		Port:          strconv.Itoa(int(cfg.DBConf.Port)),
+		User:          cfg.DBConf.User,
+		Pass:          cfg.DBConf.Pass,
+		DBName:        cfg.DBConf.DBName,
+		HidePGConfig:  false,
+		QueryTimeout:  20 * time.Minute,
+		MarketCfg:     cfg.Markets,
+		CheckedStores: true,
+		Net:           cfg.Network,
+		FeeKey:        cfg.FeePubKey,
+	}
+	storage, err := db.Open(ctxDex, pgCfg)
+	if err != nil {
+		cancel()
+		return nil, fmt.Errorf("db.Open: %v", err)
+	}
+
+	authCfg := auth.Config{
+		Storage: storage,
+		Signer:  cfg.DEXPrivKey,
+		//StartEpoch // ! different for each asset, and ambiguous
```

Comment:
I'm less sure how to handle the auth manager's start epoch setting.  It's sent to clients, so there will either need to be a value sent for each market or this value can be defined as the last start time of all the markets.

Reply:
The setting will have to be moved from a top level payload field in the `config` route response to a field in the asset objects list.

https://github.com/decred/dcrdex/blob/master/spec/fundamentals.mediawiki#exchange-variables

As far as how the start epoch is set, maybe the operator could set a single "delay" variable (`time.ParseDuration`), and the start epoch could be calculated for each market based on the market's epoch duration.

Reply:
The communications part of this issue is now in https://github.com/decred/dcrdex/issues/122

I will resolve the backend configuration part of this issue in this PR.